### PR TITLE
Application Menubar

### PR DESCRIPTION
**Pull request by sardination**
Implements a bar menu functionality. On Cocoa and supported Qt and GTK platforms, this manifests as a bar menu at the top of the screen. On all other platforms, this manifests as a menubar at the top of each window of the application.

This also implements the `webview.active_window` function that returns the active pywebview window at the time of calling.

See `examples/app_menu.py` for an implementation example.

Notes:
* Addresses and borrows a code snippet from the thread for #500
* Adjusts some basic application logic by abstracting out a `setup_app` function and also changing how GTK starts an application.

### CODE REVIEW

**Comment by r0x0r** [comment]
Great work and many thanks for tackling this. I will review and test this asap.

File: webview/__init__.py [hunk 1]
```diff
@@ -67,8 +68,10 @@
 
 token = _token
 windows = []
+menus = []
 
-def start(func=None, args=None, localization={}, gui=None, debug=False, http_server=False, user_agent=None):
+def start(func=None, args=None, localization={}, gui=None, debug=False, http_server=False, user_agent=None,
+          app_menu_list=[]):
```
Comment (r0x0r) [inline webview/__init__.py:74]: `app_menu_list=` could be just `menu=` for clarity. Also would it be possible to make a menu window not app specific? Ideally both `create_window` and `start` should accept `menu` param. The `create_window` menu would a be window specific. If both are specified, the `create_window` menu would replace the global menu.

File: webview/__init__.py [hunk 2]
```diff
@@ -120,6 +124,7 @@ def _create_children(other_windows):
         raise WebViewException('You must create a window first before calling this function.')
 
     guilib = initialize(gui)
+    guilib.setup_app()
```
Comment (r0x0r) [inline webview/__init__.py:127]: Could you combine `setup_app` this into `initialize(gui)`

File: webview/platforms/gtk.py [hunk 3]
```diff
@@ -458,14 +459,27 @@ def create_bridge():
...
     def create():
         browser = BrowserView(window)
         browser.show()
 
-    if window.uid == 'master':
+    def create_master_callback(app):
         create()
+
+    if window.uid == 'master':
+        _app.connect("activate", create_master_callback)
```
Comment (r0x0r) [inline webview/platforms/gtk.py:479]: The project uses single quotes.

**Review by r0x0r — CHANGES_REQUESTED** [review]
Thanks again for this massive effort. Please change double quotes to single ones (apart from multiline strings) and `change the app_menu_list` to `menu`. Window-specific menu can be postponed for later if it is too much work.

**Comment by r0x0r** [comment]
Cannot run `app_menu.py` with GTK, the error is 
```
(app_menu.py:37056): GLib-GIO-ERROR **: 11:21:51.525: g_menu_item_set_detailed_action: Detailed action name 'app._Test Menu_Change Active Window Content' has invalid format
```

**Comment by r0x0r** [comment]
On Windows menu appears only in the second window.

image

File: webview/__init__.py (hunk 1 above)
Comment (sardination) [reply webview/__init__.py]: I spent some time trying to get window-specific menu working on all platforms and I haven't been able to get it to work fully, so I'm going to leave that out of this PR.

**Comment by sardination** [comment]
I believe I fixed all of the noted issues, but haven't gotten around to testing it one more time on all platforms. I tried to see if I could add the per-window menu feature quickly enough, but it wasn't behaving well on all platforms, so I kept this PR to just be for an app-wide menu.

**Comment by sardination** [comment]
> I am still getting GTK error. What pygobject / GTK versions are you using?
> 
> Winforms cannot start too.

@r0x0r sorry I didn't get around to this - I'm glad it seems like you've addressed the problems. Once I have access to a Windows computer again I will test functionality.

**Pull request merged**